sources/model/entity/Event.ts

```diff
@@ -0,0 +1,175 @@
+    endTime: Date
+
+    attendees: User[] = []
+
+    @Column({name: "created_at", type: "timestamp with time zone", default: () => "CURRENT_TIMESTAMP"})
+    createdAt: Date;
+
+    @Column({name: "updated_at", type: "timestamp with time zone", default: () => "CURRENT_TIMESTAMP"})
+    updatedAt: Date;
+
+    public constructor(name: string, description:string, location: string,
+            calendar: Calendar, startTime: Date, endTime: Date) {
+        this.name = name;
+        this.description = description;
+        this.location = location;
+        this.calendar = calendar;
+        this.startTime = startTime;
+        this.endTime = endTime;
+    }
+
+    //
+    // Check if event is Valid and can be created
+    //
+    public isValid(): boolean {
+        return this.name.length > 0
+            && this.startTime <= this.endTime;
+    }
+
+     /**
+     * @brief Create an Event
+     *
+     * @param eventRepository
+     * @param eventToAttendeesRepository
+     * @param eventId
+     *
+     * @return The requested Event information on success
+     * @return null on error
+     */
+    static async createEvent(
+        eventRepository: Repository<Event>,
+        eventToAttendeesRepository: Repository<EventToAttendees>,
+        event: Event
+    ) : Promise<Event> {
+            const createdEvent = await eventRepository.save(event);
+
+            const relations = event.attendees.map(attendee => new EventToAttendees(createdEvent.id, attendee.id));
+            await eventToAttendeesRepository.save(relations);
+
+            return event;
+    }
+
+    /**
+     * @brief Get a Event (attendees list is not filled)
+     *
+     * @param eventRepository
+     * @param eventId
+     *
+     * @return The requested Event information on success
+     * @return null on error
+     */
+    static async getEventInfo(
+        eventRepository: Repository<Event>,
+        eventId: number
+    ) : Promise<Event> {
+        let event = await eventRepository
+            .createQueryBuilder("event")
+            .innerJoinAndSelect("event.owner", "owner")
+            .where({group_id: eventId})
+            .getOne();
+
+        if (!event) {
+            return null;
+        }
+
+        event.attendees = [];
+        return event;
+    }
+
+    /**
+     * @brief Get a Event with attendees list
+     *
+     * @param eventRepository
+     * @param userRepository
+     * @param eventId
+     *
+     * @return The requested Event information on success
+     * @return null on error
+     */
+    static async getEventInfoAndAttendees(
+        eventRepository: Repository<Event>,
+        userRepository: Repository<User>,
+        eventId: number
+    ) : Promise<Event> {
+        let event = await this.getEventInfo(eventRepository, eventId);
+
+        let attendees = await this.getEventAttendees(userRepository, eventId);
+        
+        event.attendees = attendees;
+        return event;
```

Comment:
Take note that this kind of request will probably change once we start using transactions

See #83